### PR TITLE
LA-240 Sort file in mySpace

### PR DESCRIPTION
**Pull request by dab246**
Pls review this guys.

https://user-images.githubusercontent.com/80730648/111935873-06c41100-8af7-11eb-9d1d-984ab4b2d5f7.mp4

### CODE REVIEW

File: data/lib/src/datasource_impl/sort_datasource_impl.dart [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+  @override
+  Future<Sorter> getSorter(OrderScreen orderScreen) {
+    final orderByStr = _sharedPreferences
+        .getString('sort_file_order_by_${orderScreen.toString()}') ?? OrderBy.modificationDate.toString();
+    final orderTypeStr = _sharedPreferences
+        .getString('sort_file_order_type_${orderScreen.toString()}') ?? OrderType.descending.toString();
+
+    var orderBy = OrderBy.modificationDate;
+    if (orderByStr == OrderBy.modificationDate.toString()) {
```
Comment (huynguyennovem) [inline data/lib/src/datasource_impl/sort_datasource_impl.dart:49]: Should replace `if` by using `switch` ?

Comment (dab246) [reply data/lib/src/datasource_impl/sort_datasource_impl.dart:49]: OrderBy.modificationDate.toString() not constant so not using switch

File: data/lib/src/datasource_impl/sort_datasource_impl.dart [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    return Future.sync(() async {
+      return Sorter(orderScreen, orderBy, orderType);
+    }).catchError((error) {
+      throw UnknownError(error.response.statusMessage);
+    });
+  }
+
+  @override
+  Future<List<T>> sortFiles<T>(List<T> listFiles, Sorter sorter) {
+    if (T == Document) {
```
Comment (hoangdat) [inline data/lib/src/datasource_impl/sort_datasource_impl.dart:73]: it is good for check instance of?

File: lib/presentation/view/context_menu/sorter_menu_action_builder.dart [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import 'package:domain/domain.dart';
+import 'package:flutter/src/foundation/key.dart';
+import 'package:flutter/src/material/list_tile.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_svg/svg.dart';
+import 'package:linshare_flutter_app/presentation/view/context_menu/context_menu_action_builder.dart';
+
+class SorterMenuTileBuilder extends ContextMenuActionBuilder<Sorter> {
+  final Sorter _sorter;
+  final bool _isSelected;
```
Comment (hoangdat) [inline lib/presentation/view/context_menu/sorter_menu_action_builder.dart:41]: strong typing this property

**Pull request closed without merging**